server/text_generation_server/models/custom_modeling/flash_cohere_modeling.py

```diff
@@ -297,8 +298,8 @@ def forward(
             # flash attention
             attn_output = attention(
                 query,
-                kv_cache[0] if SYSTEM != "ipex" else key,
-                kv_cache[1] if SYSTEM != "ipex" else value,
+                kv_cache[0] if PREFILL_IN_KV_CACHE else key,
+                kv_cache[1] if PREFILL_IN_KV_CACHE else value,
```

Comment:
Another option would be to send both to every implementation and let V2/flashinfer use the cache directly with block tables, and let V1 and other backends use the raw values.

Let's keep this for now, but maybe food for thought if this logic complexifies.

Reply:
Yeah, that crossed my mind as well, certainly worth considering in the future.